lib/full-site-editing/class-wp-rest-templates-controller.php

```diff
@@ -0,0 +1,427 @@
+			),
+			'status'      => $template->status,
+			'wp_id'       => $template->wp_id,
+		);
+
+		$response = rest_ensure_response( $result );
+		$links    = $this->prepare_links( $template->id );
+		$response->add_links( $links );
+		if ( ! empty( $links['self']['href'] ) ) {
+			$actions = $this->get_available_actions();
+			$self    = $links['self']['href'];
+			foreach ( $actions as $rel ) {
+				$response->add_link( $rel, $self );
+			}
+		}
+
+		return $response;
+	}
+
+
+	/**
+	 * Prepares links for the request.
+	 *
+	 * @since 4.7.0
+	 *
+	 * @param integer $id ID.
+	 * @return array Links for the given post.
+	 */
+	protected function prepare_links( $id ) {
+		$base = sprintf( '%s/%s', $this->namespace, $this->rest_base );
+
+		$links = array(
+			'self'       => array(
+				'href' => rest_url( trailingslashit( $base ) . $id ),
+			),
+			'collection' => array(
+				'href' => rest_url( $base ),
+			),
+			'about'      => array(
+				'href' => rest_url( 'wp/v2/types/' . $this->post_type ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get the link relations available for the post and current user.
+	 *
+	 * @return array List of link relations.
+	 */
+	protected function get_available_actions() {
+		$rels = array();
+
+		$post_type = get_post_type_object( $this->post_type );
+
+		if ( current_user_can( $post_type->cap->publish_posts ) ) {
+			$rels[] = 'https://api.w.org/action-publish';
+		}
+
+		if ( current_user_can( 'unfiltered_html' ) ) {
+			$rels[] = 'https://api.w.org/action-unfiltered-html';
+		}
+
+		return $rels;
+	}
+
+	/**
+	 * Retrieves the query params for the posts collection.
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		$query_params = parent::get_collection_params();
+
+		$query_params['theme'] = array(
+			'description' => __( 'Limit to the current theme templates.', 'gutenberg' ),
+			'type'        => 'string',
+		);
+		$query_params['wp_id'] = array(
+			'description' => __( 'Limit to the specified post id.', 'gutenberg' ),
+			'type'        => 'integer',
+		);
+
+		return $query_params;
+	}
+
+	/**
+	 * Retrieves the block type' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => $this->post_type,
+			'type'       => 'object',
+			'properties' => array(
+				'id'          => array(
+					'description' => __( 'ID of template.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'slug'        => array(
+					'description' => __( 'Unique slug identifying the template.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
```

Comment:
```suggestion
```

Since we use this in creation, we need to drop this. We don't yet support a `createonly` parameter unfortunately.